### ISSUE TITLE
State during debugging

### ISSUE THREAD

**Issue by sharwell** [tunnelvisionlabs/JustMyCodeToggle#6]
1. When this command is used during a debugging session, are the changes applied immediately or only after the next time a debugging session is started?
1. Should this command be disabled during a debugging session? I noticed the option under Tools &rarr; Options... remains enabled, but there may be usability benefits to disabling the toggle command when it will not be functional.

**Comment by jnm2** [comment]
It's overwhelmingly common for me to need to toggle it either direction during a debug session. That's how the current debug options checkbox works.

I can't imagine a scenario in which I would be happy that it's disabled. If I'm not debugging, I'm pressing a keyboard shortcut and expecting the JMC state to toggle immediately and have an effect on the very next debug session. If I am debugging, I'm pressing a keyboard shortcut and expecting the JMC state to toggle and have an immediate effect like it does in the debug options window. That's about as helpful as it can get IMO.

If I'm using a keyboard shortcut and the item is disabled, I won't know that and it will be irritating when the debug session starts without it. In many cases I'll have to restart the debugging session from the beginning and sometimes that is time consuming.

If I'm using the right click menu and the item is disabled, at least I'll see it and know what to expect, but it'll be nonetheless irritating because I was going to do it my way and you're making me do it in an arbitrary order. Plus in some cases it's nice to have JMC off in the first few milliseconds of the debug session starting- a feat that is hard to reliably achieve even with a keyboard shortcut. So I'll have to press F10, wait for debugging to start, then toggle JMC, then press F5.

tl;dr seems simpler to mimick the existing debug options behavior: take immediate effect, never disable

**Comment by sharwell** [comment]
@jnm2 Thanks. I've never toggled this command myself so my questions were based on not knowing how the existing option behaved if you did toggle it while debugging. Seems easy enough to not ever disable it.
